common/src/main/java/org/apache/nemo/common/DirectByteBufferOutputStream.java

```diff
@@ -31,11 +32,10 @@
 public final class DirectByteBufferOutputStream extends OutputStream {
```

Comment:
When is `dataList` deleted and when is the memory released, especially since the memory used is allocated off heap?
Do we still comply with the `DataPersistenceProperty`? (i.e., delete intermediate data once used)
Can you add comments perhaps in this class and others such as `SerializedPartition` to clarify this?

Reply:
Yes, it also depends on the `DataPersistenceProperty`.

Reply:
Sounds good!